naucse/routes.py

```diff
@@ -122,16 +122,23 @@ def prv_nxt_teller(run, lesson):
         for material in session.materials
         if material.lesson
     ]
+
+    prv_link, prv_title, nxt_link, nxt_title = None, None, None, None
+
     for prev, current, next in zip([None] + lessons,
                                    lessons,
                                    lessons[1:] + [None]):
         if current.slug == lesson.slug:
             if prev:
                 prev = prev.index_page
+                prv_link = str(prev)
```

Comment:
This shouldn't return the text representation of a page, but an URL. The text representation is a human-readable string that might easily change in the future.
Make `prv_next_teller` take the `subpage_url` function as argument if necessary.